onnxruntime/core/platform/windows/device_discovery.cc

```diff
@@ -362,90 +362,109 @@ std::unordered_map<uint64_t, DeviceInfo> GetDeviceInfoD3D12() {
   return device_info;
 }
 
+typedef HRESULT(WINAPI* PFN_DXCoreCreateAdapterFactory)(REFIID riid, void** ppvFactory);
```

Comment:
Hide the whitespace diffs as there are very few actual lines that are changed. 

<img width="320" alt="image" src="https://github.com/user-attachments/assets/5a24674a-ad1d-48bc-a688-3bb4d1375eeb" />